File: app/src/main/java/tpcreative/co/qrscanner/common/entities/SaveDao.java
package tpcreative.co.qrscanner.common.entities;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

@Dao
public interface SaveDao {
    @Insert
    void insert(SaveEntity... saves);

    @Update
    void update(SaveEntity... saves);

    @Delete
    void delete(SaveEntity... saves);

    @Query("DELETE FROM save")
    public void deleteAllItems();

    @Query("DELETE FROM save where uuId =:uuId")
    public void deleteSpecific(String uuId);

    @Query("Select * FROM save ORDER BY updatedDateTime DESC")
    List<SaveEntity> loadAll();

    @Query("Select * FROM save where isSynced =:isSynced ORDER BY updatedDateTime DESC")
    List<SaveEntity> loadAll(boolean isSynced);

    @Query("Select * FROM save WHERE contentUnique = :contentUnique")
    SaveEntity loadItem(String contentUnique);
}

File: app/src/main/java/tpcreative/co/qrscanner/common/entities/SaveDao.kt
package tpcreative.co.qrscanner.common.entities
import androidx.room.*

@Dao
interface SaveDao {
    @Insert
    fun insert(vararg saves: SaveEntity?)
    @Update
    fun update(vararg saves: SaveEntity?)
    @Delete
    fun delete(vararg saves: SaveEntity?)
    @Query("DELETE FROM save")
    fun deleteAllItems()
    @Query("DELETE FROM save where uuId =:uuId")
    fun deleteSpecific(uuId: String?)
    @Query("Select * FROM save ORDER BY updatedDateTime DESC")
    fun loadAll(): MutableList<SaveEntity>?
    @Query("Select * FROM save where isSynced =:isSynced ORDER BY updatedDateTime DESC")
    fun loadAll(isSynced: Boolean): MutableList<SaveEntity>?
    @Query("Select * FROM save WHERE contentUnique = :contentUnique")
    fun loadItem(contentUnique: String?): SaveEntity?
    @Query("Select * FROM save WHERE id = :id")
    fun loadItem(id: Int?): SaveEntity?
    @Query("Select * FROM save WHERE favorite = :isFavorite")
    fun loadAllItem(isFavorite : Boolean): MutableList<SaveEntity>?
}